### PR TITLE
doc: Update `backpressuring-in-streams.md` by introducing `pipeline` since v10.0.0

### PR DESCRIPTION
**Pull request by ghost**
Ref: https://github.com/nodejs/nodejs.org/issues/1668

PS：Considering the method is introduced since v10.0.0, and we can still use chains of `pipe()` together, so I still keep the original codes and explainations. But just add for new introductions of `pipeline()`.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
LGTM. I’ll soon be releasing readable-stream@3 that enables pipeline in Node 6+.
We should be updating this accordingly.

**Comment by sonicdoe** [comment]
If [`stream.pipeline()`](https://nodejs.org/api/stream.html#stream_stream_pipeline_streams_callback) fully replaces [`pump`](https://github.com/mafintosh/pump), I’d suggest rewording the section so that it is introduced first and to mention that `pump` is only necessary for Node.js v8 or earlier.

What do you think? If you’d like, I can push a small example.

File: locale/en/docs/guides/backpressuring-in-streams.md [hunk 1]
```diff
@@ -88,6 +88,68 @@ a chunk of data were to fail to be properly received, the `Readable` source or
 properly destroy all the streams in a pipeline if one of them fails or closes,
 and is a must have in this case!
 
+As for Node 10.0.0 or later version, [`pipeline`][] is introduced to replace for
+[`pump`][]. This is a module method to pipe between streams forwarding errors
+and properly cleaning up and provide a callback when the pipeline is complete.
+
+You can use it like this following:
```
Comment (Bamieh) [inline locale/en/docs/guides/backpressuring-in-streams.md]: to stay consistent with other parts of the document, lets rephrase this:

> Here is an example of using pipeline:

File: locale/en/docs/guides/backpressuring-in-streams.md [hunk 2]
```diff
@@ -88,6 +88,68 @@ a chunk of data were to fail to be properly received, the `Readable` source or
...
+  (err) => {
+    if (err) {
+      console.error('Pipeline failed', err);
+    } else {
+      console.log('Pipeline succeeded');
+    }
+  }
+);
+```
+or if you wanna `async` with `await`, you can use it wrapped by [`promisify`][]:
```
Comment (Bamieh) [inline locale/en/docs/guides/backpressuring-in-streams.md]: Let's replace "wanna" since it is not formal english

> You can call [`promisify`][] on `pipeline` to use it with `async/await`:

File: locale/en/docs/guides/backpressuring-in-streams.md [hunk 3]
```diff
@@ -88,6 +88,68 @@ a chunk of data were to fail to be properly received, the `Readable` source or
...
+     fs.createReadStream('The.Matrix.1080p.mkv'),
+     zlib.createGzip(),
+     fs.createWriteStream('The.Matrix.1080p.mkv.gz'),
+  );
+}
+
+// Either of the following ways you can call the `run()`:
+
+// Way 1: You can use this to catch exceptions in async mode.
+run().catch((err) => console.log('Pipeline failed', err));
```
Comment (Bamieh) [inline locale/en/docs/guides/backpressuring-in-streams.md]: I feel this is redundant, maybe just having the following code snippet is enough:

```js
async function run() {
  try {
    await pipeline(
       fs.createReadStream('The.Matrix.1080p.mkv'),
       zlib.createGzip(),
       fs.createWriteStream('The.Matrix.1080p.mkv.gz'),
    );
  } catch(err) {
    console.error('Pipeline failed', err);
  }
}
```

**Review by Bamieh — CHANGES_REQUESTED** [review]
Thank you for taking the time to improve this

**Comment by Bamieh** [comment]
Closes https://github.com/nodejs/nodejs.org/issues/1668

File: locale/en/docs/guides/backpressuring-in-streams.md [hunk 4]
```diff
@@ -88,6 +88,68 @@ a chunk of data were to fail to be properly received, the `Readable` source or
...
+You can use it like this following:
+
+```javascript
+const { pipeline } = require('stream');
+const fs = require('fs');
+const zlib = require('zlib');
+
+// Use the pipeline API to easily pipe a series of streams
+// together and get notified when the pipeline is fully done.
+// A pipeline to gzip a potentially huge tar file efficiently:
```
Comment (vsemozhetbyt) [inline locale/en/docs/guides/backpressuring-in-streams.md]: `tar file` -> `video file`?

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 1 above)
Comment (ghost) [reply locale/en/docs/guides/backpressuring-in-streams.md]: Fixed

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 3 above)
Comment (ghost) [reply locale/en/docs/guides/backpressuring-in-streams.md]: Fixed

File: locale/en/docs/guides/backpressuring-in-streams.md (hunk 4 above)
Comment (ghost) [reply locale/en/docs/guides/backpressuring-in-streams.md]: Fixed

**Comment by ghost** [comment]
@vsemozhetbyt, @sonicdoe & @Bamieh：
All have been fixed according to your ideas :)

@sonicdoe：The `pump` is a 3-rd party tool with an example of a link, and if you click that it will guide you there (https://github.com/mafintosh/pump). Since there's already a link direct to GitHub with an example there. And I'm not sure whether we can allow a 3-rd party example directly in Nodejs? If you are glad to do this, maybe you can submit another PR after mine.

@Bamieh：Maybe when this submit is merged, you can close your issue, or I'll help to close.

**Review by vsemozhetbyt — APPROVED** [review]

**Review by daxlab — APPROVED** [review]

**Review by Bamieh — APPROVED** [review]

**Comment by ghost** [comment]
@mcollina：I've merged this, and waiting for your changes onto my submitted one. Then I'll call other translators to make a translation on what we've changed.

**Comment by mcollina** [comment]
I think you can go ahead and start translating this change. It might take me a bit more time, and I do not want to stop any activity.

**Comment by ghost** [comment]
OK. Thanks! @mcollina
